### PR TITLE
Add a check for invalid/junk messages

### PR DESCRIPTION
**Pull request by sanimej**
Relates to [docker/docker #20350] (https://github.com/docker/docker/issues/20350)

Signed-off-by: Santhosh Manohar <santhosh@docker.com>

### CODE REVIEW

File: resolver.go [hunk 1]
```diff
@@ -200,6 +200,9 @@ func (r *resolver) ServeDNS(w dns.ResponseWriter, query *dns.Msg) {
 		err  error
 	)
 
+	if len(query.Question) == 0 {
```
Comment (mavenugo) [inline resolver.go]: Can you pls also check if query is nil ?

Comment (sanimej) [inline resolver.go]: query can't be nil; miekg/dns library calls ServeDNS with a new dns.Msg. But doesn't hurt to add defensive check..

**Comment by mavenugo** [comment]
LGTM

**Comment by aboch** [comment]
LGTM

**Pull request merged**